### ISSUE TITLE
[SPDesignerSettings] Web Application URL not stardardized / fail-proof

### ISSUE THREAD

**Issue by MakwaWes** [PowerShell/SharePointDsc#689]
**Details of the scenario you try and problem that is occurring:**
While working with the resource "SPDesignerSettings", I found out that if you specify the URL of the web application in capitals, the resource will fail.
Also, if your web application is using a custom port, you can specify it in the resource but if you are using the standard ports "80" or "443", you must not specify the port in the URL.

It seems related to the way the cmdlet **Get-SPWebApplication** returns the URL in lowercase, without standard port and the filter used on it :
```
#Check if web application exists
$webapp = Get-SPWebApplication | Where-Object -FilterScript { 
 ($_.Url).StartsWith($params.Url) 
}
```
[Line 69 in the code](https://github.com/PowerShell/SharePointDsc/blob/5cf5c6678d7e03e0ee79212b907eb025ff95901b/Modules/SharePointDsc/DSCResources/MSFT_SPDesignerSettings/MSFT_SPDesignerSettings.psm1#L69)

**Example**
The web application URL : **http://MYWEBAPPURL:8089**
Fails :
```
SPDesignerSettings MyDesignerSettings {
Url = 'http://MYWEBAPPURL:8089'
SettingsScope = 'WebApplication'
AllowSharePointDesigner = $true
}
```
Works :
```
SPDesignerSettings MyDesignerSettings {
Url = 'http://mywebappurl:8089'
SettingsScope = 'WebApplication'
AllowSharePointDesigner = $true
}
```

The web application URL : **http://mywebappurl:443**
Fails :
```
SPDesignerSettings MyDesignerSettings {
Url = 'http://mywebappurl:443
SettingsScope = 'WebApplication'
AllowSharePointDesigner = $true
}
```
Works :
```
SPDesignerSettings MyDesignerSettings {
Url = 'http://mywebappurl'
SettingsScope = 'WebApplication'
AllowSharePointDesigner = $true
}
```
Thanks for the nice SharePoint DSC resource !

Regards,
Wes.

**Comment by ykuijs** [comment]
Hi @MakwaWes, thanks for reporting. I have marked this as a bug.

**Comment by ykuijs** [comment]
StartsWith is indeed case sensitive. We need to update this to StartsWith($params.Url, "CurrentCultureIgnoreCase").

I will include this fix in a next PR

**Comment by ykuijs** [comment]
Fixes implemented in PR #695

**Comment by MakwaWes** [comment]
Thanks @ykuijs !
👍
